pkg/core/transactor/transactor.go

```
@@ -114,6 +115,27 @@ func (t *Transactor) Listen() {
 	}
 }
 
+func (t *Transactor) canStake() bool {
+	if t.getSyncProgress() == 100 {
+		return true
+	}
+
+	// Check for our sync progress three more times. If we still can't stake after
+	// the third check, it's better to just return false, and give the Transactor
+	// control back over the Listen goroutine.
+	interval := 5 * time.Second
+
+	for i := 0; i < 3; i++ {
+		time.Sleep(interval * time.Duration(i+1))
+
+		if t.getSyncProgress() == 100 {
+			return true
+		}
+	}
```

I think that this kind of retry is too naìve. It is fine to perform a number of attempts, but the wallet needs to get  feeback rather allow for silent failure. It might make sense to use the RPCBus here and return feedback to the wallet

---

Well, using the RPCBus here to fetch the sync progress really doesn't change much, besides the infrastructure we're using (unless I'm missing something). I can of course add logging, though.